src/helpers/getByAPI.js

```diff
@@ -1,24 +1,44 @@
 // @flow
 import * as React from 'react';
+import prettyFormat from 'pretty-format';
 import ErrorWithStack from './errorWithStack';
 
+const getNodeByType = (node, type) => node.type === type;
+
 const getNodeByName = (node, name) =>
-  node.type.name === name ||
-  node.type.displayName === name ||
-  node.type === name;
+  typeof node.type !== 'string' &&
+  (node.type.displayName === name || node.type.name === name);
+
+const getNodeByText = (node, text) => {
+  try {
+    // eslint-disable-next-line
+    const { Text, TextInput } = require('react-native');
+    return (
+      (getNodeByType(node, Text) || getNodeByType(node, TextInput)) &&
+      (typeof text === 'string'
+        ? text === node.props.children
+        : text.test(node.props.children))
+    );
+  } catch (error) {
+    throw new Error(
+      `Currently the only supported library to search by text is \`react-native\`.\n\n${error}`
```

Comment:
Can you extract constructing this string to a method? Or even create custom Error?